micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jetty/JettyStatisticsMetrics.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2017 Pivotal Software, Inc.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.core.instrument.binder.jetty;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.TimeGauge;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+
+import org.eclipse.jetty.server.handler.StatisticsHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.ToDoubleFunction;
+import java.util.function.ToLongFunction;
+
+public class JettyStatisticsMetrics implements MeterBinder {
+    private final StatisticsHandler statisticsHandler;
+    private static final List<Tag> EMPTY = new ArrayList<>();
+
+    public JettyStatisticsMetrics(StatisticsHandler statisticsHandler) {
+        this.statisticsHandler = statisticsHandler;
+    }
+
+    @Override
+    public void bindTo(MeterRegistry reg) {
+        bindTimer(reg,"jetty.requests", "Request duration", StatisticsHandler::getRequests, StatisticsHandler::getRequestTimeTotal);
+        bindTimer(reg,"jetty.dispatched", "Dispatch duration", StatisticsHandler::getDispatched, StatisticsHandler::getDispatchedTimeTotal);
+
+        bindCounter(reg,"jetty.async.requests", "Total number of async requests", StatisticsHandler::getAsyncRequests);
+        bindCounter(reg,"jetty.async.dispatches", "Total number of requested that have been asynchronously dispatched", StatisticsHandler::getAsyncDispatches);
+        bindCounter(reg,"jetty.async.expires", "Total number of async requests requests that have expired", StatisticsHandler::getExpires);
+        bindCounter(reg,"jetty.responses.bytes", "Total number of bytes across all responses", StatisticsHandler::getResponsesBytesTotal);
```

Comment:
Could we do something relatively unobtrusive to create a function-tracking summary? Effectively, if you use this operationally you want to look at byte _throughput_, not a total accumulation of bytes. Naturally Prometheus generates rates from accumulated values, but we need to think past just Prometheus.

Reply:
Then we should move "bytes" to the base unit rather than as part of the name. The remaining metric name will have to be different to disambiguate it with the status counter.

Reply:
I'm not sure how that could work out. There might be a way to add a connector that records bytes written as a T function. I can't seem to find a good example, and I'm kind of scared to implement it, since I don't really know that much about Jetty internals.

Reply:
Disregard the summary thing. I'm not thinking straight. Byte throughput is already achieved by the function counter. Still can move `bytes` to the base unit.